js/ui/search.js

```diff
@@ -414,6 +414,13 @@ var SearchResults = class {
         Util.blockClickEventsOnActor(this.actor);
 
         this._parentalControlsManager = ParentalControlsManager.getDefault();
+        this._parentalControlsManager.connect('changed', () => {
+            this._reloadRemoteProviders();
+
+            // Reload internet search providers too
+            this._unregisterProvider(this._internetProvider);
+            this._registerProvider(this._internetProvider);
```

Comment:
I believe you also need to reload the internetProvider here. What it does today is that it loads the internetProvider on constructor using `InternetSearch.getInternetSearchProvider` and that always return the current default browser, but if the default browser is disabled but we have other browsers enabled we end up with no internet search provider.

This should be fine once we support disabling all browsers at once but I think this code could still ignore this assumption and check all available browsers (starting with the default one) and whether they are blacklisted, something like (pseudocode):
```

reloadInternetProvider() {
  if (this._internetProvider)
    this._unregisterProvider(this._internetProvider);
  internetProviders = InternetSearch.getInternetSearchProviders();
  for (let i = 0; i < internetProviders.length; i++) {
    let internetProvider = internetProviders[i];
    if (this._registerProvider(internetProvider))
      this._internetProvider = internetProvider;
  }
}
```

Also change _registerProvider to return false if the provider is blacklisted and true otherwise and invoke reloadInternetProvider() from the constructor.

You will also need to change or add a new function to get all available internet search providers (right now InternetSearch.getInternetSearchProvider() only returns the default browser). To get all browsers use Gio.get_all_for_type('x-scheme-handler/http') - see also js/misc/utils.js:getBrowserApp|Id methods.

Reply:
Note that the default browser should still be considered first here, so for example an impl of `InternetSearch.getInternetSearchProviders()` as mentioned above should return the default browser as first on the list.

Reply:
I am not sure if the order is important here (if it changes the way the results are displayed), but if so you also need to reload the internet search provider before reloading the remote providers (as done in the constructor).

Reply:
I confirm with @wjt and this is not the approach we are taking here.

GNOME settings UI will expose all-or-nothing deal for browsers and hence disable everything for `x-scheme-handler/http` (including the default InternetProvider). This makes the shell's bit(this PR) easy to process.